cmake/CMakeLists.txt

```diff
@@ -16,7 +16,11 @@ endif()
 project(protobuf C CXX)
 
 # Add c++11 flags
-set(CMAKE_CXX_STANDARD 11)
+if (CYGWIN)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11")
+else()
+  set(CMAKE_CXX_STANDARD 11)
+endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
Can you move all of CMAKE_CXX_STANDARD/CMAKE_CXX_STANDARD_REQUIRED/CMAKE_CXX_EXTENSIONS inside the same else group?

Reply:
Fixed!